### PR TITLE
Add documentation and examples for COPY options

### PR DESCRIPTION
**Pull request by andreasscherbaum**

### CODE REVIEW

**Comment by andreasscherbaum** [comment]
This PR is WIP, and not yet complete.

@mkiyama Can you please see if the format is ok, or if I added mistakes somewhere?

Will add more options soon.

File: gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml [hunk 1]
```diff
@@ -9,9 +9,61 @@
...
+    <title>Loading From a File</title>
+    <body>
+      <p>The <codeph>COPY</codeph> command asks the <codeph>postgres</codeph> backend to open the
+         specified file, read it and append it to the table. In order to be able to read the file,
+         the backend needs to have read permissions on the file, and the file name must be
+         specified using an absolute path on the master host, or a relative path to the master
+         data directory.</p>
+      <p>
+          <codeblock>COPY <i>table_name</i> FROM <i>/path/to/filename</i>;</codeblock>
+      </p>
```
Comment (mkiyama) [inline gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml]: need a \</body> before the \</topic>

File: gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml [hunk 2]
```diff
@@ -9,9 +9,61 @@
...
+    <title>Loading From STDIN</title>
+    <body>
+      <p>To avoid the problem of copying the data file to the master host before loading the data,
+         <codeph>COPY FROM STDIN</codeph> uses the Standard Input channel and feeds data directly
+         into the <codeph>postgres</codeph> backend. After the <codeph>COPY FROM STDIN</codeph>
+         command started, the backend will accept lines of data until a single line only contains
+         a backslash-period (\.).</p>
+      <p>
+          <codeblock>COPY <i>table_name</i> FROM <i>STDIN</i>;</codeblock>
+      </p>
```
Comment (mkiyama) [inline gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml]: need a \</body> before the \</topic>

File: gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml [hunk 3]
```diff
@@ -9,9 +9,61 @@
...
+         the host where the psql client runs, and must be accessible to the user which runs the
+         client.
+         To avoid the problem of copying the data file to the master host before loading the data,
+         <codeph>COPY FROM STDIN</codeph> uses the Standard Input channel and feeds data directly
+         into the <codeph>postgres</codeph> backend. After the <codeph>COPY FROM STDIN</codeph>
+         command started, the backend will accept lines of data until a single line only contains
+         a backslash-period (\.).</p>
+      <p>
+          <codeblock>\copy <i>table_name</i> FROM <i>filename</i>;</codeblock>
+      </p>
```
Comment (mkiyama) [inline gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml]: need a \</body> before the \</topic>

File: gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml [hunk 4]
```diff
@@ -9,9 +9,61 @@
...
+  </topic>
+  <topic id="topic4" xml:lang="en">
+    <title>Input Format</title>
+    <body>
+      <p><codeph>COPY FROM STDIN</codeph> accepts a <codeph>FORMAT</codeph> parameter, which
+         specifies the format of the input data. The possible values are <codeph>TEXT</codeph>,
+         <codeph>CSV</codeph> (Comma Separated Values), and <codeph>BINARY</codeph>.</p>
+      <p>
+          <codeblock>COPY <i>table_name</i> FROM <i>/path/to/filename</i> WITH FORMAT csv;</codeblock>
+      </p>
```
Comment (mkiyama) [inline gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml]: need a \</body> before the \</topic>

**Comment by andreasscherbaum** [comment]
Added missing </copy> tags

**Comment by hlinnaka** [comment]
Is this ready to be committed now?

**Comment by andreasscherbaum** [comment]
Uh, forgot about this one. Will update it with more examples.

**Comment by ashwinstar** [comment]
@andreasscherbaum Seems stuck again, can we bring this to closure.

**Comment by danielgustafsson** [comment]
And another ping @andreasscherbaum

File: gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml [hunk 5]
```diff
@@ -9,9 +9,66 @@
...
+         To avoid the problem of copying the data file to the master host before loading the data,
+         <codeph>COPY FROM STDIN</codeph> uses the Standard Input channel and feeds data directly
+         into the <codeph>postgres</codeph> backend. After the <codeph>COPY FROM STDIN</codeph>
+         command started, the backend will accept lines of data until a single line only contains
+         a backslash-period (\.).</p> psql is wrapping all of this into the handy
+         <codeph>\copy</codeph> command.
+      <p>
+          <codeblock>\copy <i>table_name</i> FROM <i>filename</i>;</codeblock>
+      </p>
+   </body>
```
Comment (mkiyama) [inline gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml]: @andreasscherbaum 
dita markup is not quite correct in these lines.

```
+     </p> psql is wrapping all of this into the handy
+         <codeph>\copy</codeph> command.
+      <p>
+          <codeblock>\copy <i>table_name</i> FROM <i>filename</i>;</codeblock>
+      </p>
```
I think this is what you want.

```
+      psql is wrapping all of this into the handy
+         <codeph>\copy</codeph> command.</p> 
+      
+          <codeblock>\copy <i>table_name</i> FROM <i>filename</i>;</codeblock>
+     
```

**Comment by mkiyama** [comment]
@dyozie 
Doc update for Admin Guide

File: gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml (hunk 5 above)
Comment (andreasscherbaum) [reply gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml]: Forgot the missing </p> - added it now and will commit with the next changes. Thanks for spotting this!

**Comment by andreasscherbaum** [comment]
Added more examples for the input format. How does it look like?

File: gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml [hunk 6]
```diff
@@ -9,9 +9,78 @@
...
+      <p>The <codeph>FORMAT csv</codeph> will read comma-separated values. The
+         <codeph>FORMAT text</codeph> by default uses tabulators to separate the values, the
+         <codeph>DELIMITER</codeph> option specifies a different character as value delimiter.</p>
+      <p>
+          <codeblock>COPY <i>table_name</i> FROM <i>/path/to/filename</i> WITH (FORMAT text, DELIMITER '|');</codeblock>
+      </p>
+      <p>By default, the default client encoding is used, this can be changed with the
+         <codeph>ENCODING</codeph> option. This is useful if data is coming from another operating
+         system.
+      <p>
```
Comment (mkiyama) [inline gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml]: Similar mismatched paragraph tag  here.
You need to either remove the \<p> or add a \</p>

File: gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml [hunk 7]
```diff
@@ -9,9 +9,78 @@
...
+      <p>Do not confuse the psql <codeph>\copy</codeph> command with the <codeph>COPY</codeph> SQL
+         command. The <codeph>\copy</codeph> invokes a regular <codeph>COPY FROM STDIN</codeph>
+         and sends the data from the psql client to the backend. Therefore any file must reside on
+         the host where the psql client runs, and must be accessible to the user which runs the
+         client.</p>
+      <p>To avoid the problem of copying the data file to the master host before loading the data,
+         <codeph>COPY FROM STDIN</codeph> uses the Standard Input channel and feeds data directly
+         into the <codeph>postgres</codeph> backend. After the <codeph>COPY FROM STDIN</codeph>
+         command started, the backend will accept lines of data until a single line only contains
+         a backslash-period (\.).</p> psql is wrapping all of this into the handy
```
Comment (mkiyama) [inline gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml]: You need to either remove a \</p> or add a \<p>

Either this
```
 a backslash-period (\.). psql is wrapping all of this into the handy
```
or this
```
 a backslash-period (\.). </p><p>psql is wrapping all of this into the handy

File: gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml (hunk 6 above)
Comment (andreasscherbaum) [reply gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml]: Fixed

File: gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml (hunk 7 above)
Comment (andreasscherbaum) [reply gpdb-doc/dita/admin_guide/load/topics/g-loading-data-with-copy.xml]: Fixed

**Comment by andreasscherbaum** [comment]
Any objections that I push this?

**Comment by dyozie** [comment]
Looks good - if there are any dita formatting errors we can deal with them after the merge.

**Pull request merged**